### PR TITLE
Fix bug for Milling non-count_by_charges items

### LINKED ISSUE #1715: Chitin milling, resultant chitin powder is weightless and crashes game.
### Describe the bug

If you mill regular or borosilicated chitin in a windmill the resulting chitin powder with have a count of 0 and crash the game if you try to pick it up and put it down. If you load the game after milling and **before** collecting the results you get the proper amount of chitin powder. If you take the 0 count powder out **then** save/load it will become 4 chitin powder no matter how many chunks had gone into the mill.

And the mill says there's 0 chitin chunks inside for every chunk you put it, volume seems right though.

### Steps To Reproduce

1. Place windmill
2. Load 1 or more chunks of chitin into windmill, noting number
3. start mill
4. wait 6 hours

5a. remove product
6a. it's 0 count of chitin powder
7a. save & load game
8a. now it's 4 count of chitin powder

5b. save & load before removing chitin powder
6b. remove product
7b. it's the proper number of chitin powder

### Expected behavior

It always being the right amount.

### Screenshots

image

image

image

image

with 10 chunks milled, collected after save/load
image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 5d36160 [64-bit]
- Graphics Version: Tiles
- Game Language:  []
- Mods loaded: [
    Bright Nights [dda],
    Disable NPC Needs [no_npc_food],
    Simplified Nutrition [novitamins],
    Elevated bridges [elevated_bridges]
]

### Additional context

Related to PR #1682

### PR DESCRIPTION
**Pull request by KheirFerrum**
#### Summary

SUMMARY: Bugfixes "Fixes milling chitin chunks into powder creating crashes."

#### Purpose of change

Fix #1715. Updates PR #1682.

#### Describe the solution

Causes milling results to take into account if the starting material is not charge based. Update documentation to reflect limitations yet to be overcome, update mill description to reflect expanded mill usage.

#### Describe alternatives you've considered

- Make it work better.
Maybe another time? T^T

#### Testing

Add chitin to the windmill, check that it displays properly. Wait 6 hours (~get mauled by cougar~), check that materials _appear_ correct when inspecting. Get annoyed that the unload message still sees them as separate stacks, but can confirm that products are all correct.

#### Additional context

This could actually be updated by having the game tally up all the materials of non-charge (and all charges of the same type) and directly add a single material with that much charge (or in the case of non-charge items, create that many items). I didn't do this because that code feels like a nightmare to write and test. Maybe another time.

### CODE REVIEW

File: doc/JSON_INFO.md [hunk 1]
```diff
@@ -1635,7 +1635,7 @@ See also VEHICLE_JSON.md
     [ "45", [ "m1911mag", "m1911bigmag" ] ],
 ],
 "milling": {                                 // Optional. If given, the item can be milled in a water/wind mill.
-  "into": "flour",                           // The item id of the product.
+  "into": "flour",                           // The item id of the product. Product MUST be something that uses charges.
```
Comment (olanti-p) [inline doc/JSON_INFO.md:1638]: If it's important that product uses charges, it needs to be checked in `Item_factory::check_definitions()`. People _will_ miss this note, even with the caps, but they can't miss the game screaming at them in debugmsgs.

File: src/iexamine.cpp [hunk 2]
```diff
@@ -5710,13 +5710,19 @@ void iexamine::quern_examine( player &p, const tripoint &examp )
...
                     if( it.typeId() == itype_fake_milling_item ) {
-                        pop += "\n" + colorize( _( "You see some grains that are not yet milled to fine flour." ),
+                        pop += "\n" + colorize( _( "You see that the milling process is not yet complete." ),
                                                 c_red ) + "\n";
                         continue;
                     }
-                    pop += "-> " + item::nname( it.typeId(), it.charges ) + " (" + std::to_string( it.charges ) + ")\n";
+                    mill_list[it] += it.count();
+                }
+                for( auto it_mill = mill_list.begin(); it_mill != mill_list.end(); ++it_mill ) {
```
Comment (olanti-p) [inline src/iexamine.cpp:5722]: Clang-tidy complains
```
/home/runner/work/Cataclysm-BN/Cataclysm-BN/src/iexamine.cpp:5722:17: error: use range-based for loop instead [modernize-loop-convert,-warnings-as-errors]
                for( auto it_mill = mill_list.begin(); it_mill != mill_list.end(); ++it_mill ) {
                ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                   (auto & it_mill : mill_list)
```

File: doc/JSON_INFO.md (hunk 1 above)
Comment (KheirFerrum) [reply doc/JSON_INFO.md:1638]: Having trouble doing this. I might just spend the time to make it accept non-charge products...

Comment (olanti-p) [reply doc/JSON_INFO.md:1638]: What trouble are you having? `itype` provides `count_by_charges()` method, it should be a matter of adding
```
else if ( !type->milling_data->into_->count_by_charges() ) {
    msg += "type to mill into must be counted by charges: " + type->milling_data->into_.str() + "\n";
}
```
after the check added in previous PR.

**Comment by Coolthulhu** [comment]
Still needs that warning on load.

File: doc/JSON_INFO.md (hunk 1 above)
Comment (KheirFerrum) [reply doc/JSON_INFO.md:1638]: Sorry, I was writing the code wrong, I think I searched for a template instead.

**Review by olanti-p — APPROVED** [review]

**Pull request merged**